### PR TITLE
Make CdpDataSource GetDatasetsMetadataAsync public and return metadata

### PR DESCRIPTION
**Pull request by LucGenetier**

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Connectors/Tabular/Services/InternalObjects.cs [hunk 1]
```diff
@@ -37,24 +37,37 @@ internal class RawTable
...
+    public class DatasetMetadata
     {
+        [JsonConstructor]
+        public DatasetMetadata(MetadataTabular tabular, MetadataBlob blob, string datasetFormat, IReadOnlyCollection<MetadataParameter> parameters)
+        {
+            Tabular = tabular;
+            Blob = blob;
+            DatasetFormat = datasetFormat;
+            Parameters = parameters;
+        }
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Connectors/Tabular/Services/InternalObjects.cs:49]: do we need this ctor? All the properties already have setters?

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**